src/main/java/com/diffmin/patch/PatchApplication.java

```diff
@@ -77,6 +78,29 @@ private static void performInsertion(
                                 ((CtExecutable<?>) toBeInserted.getParent()).getThrownTypes());
                 ((CtExecutable<?>) inWhichElement).setThrownTypes(thrownTypesCopy);
                 break;
+            case CONTAINED_TYPE:
+                // Inserting into CtPackage
+                ((CtPackage) inWhichElement).addType((CtType<?>) toBeInserted.clone());
+
+                // Inserting into CtCompilationUnit
+                List<CtCompilationUnit> compilationUnits =
+                        List.copyOf(
+                                ((CtPackage) inWhichElement)
+                                        .getDeclaringModule()
+                                        .getFactory()
+                                        .CompilationUnit()
+                                        .getMap()
+                                        .values());
```

Comment:
I plan to put this logic in `SpoonUtil` so that it can be reused.

Reply:
Yes, this should be a method along the lines of `getAllCompilationUnits(CtElement element)`. The implementation can then look like `element.getParent(CtModule.class).getFactory().CompilationUnit().getMap().values()`.

It could be further abstracted into `getUnambiguousCompilationUnit(CtElement)`, where it fetches only one compilation unit, and raises if there's anything but one.